packages/react-components/src/hooks/useAlarms/utils/queryUtils.ts

```diff
@@ -1,6 +1,28 @@
 import { UseQueryResult } from '@tanstack/react-query';
 import { AlarmDataStatus } from '../types';
 
+/**
+ * Combine two query statuses
+ *
+ * If any one status isLoading/Refetching/Error is true then the running status is also true
+ * If any one status isSuccess is false then running status is also false
+ * Combine all errors into a running list
+ */
+// Combine two query statuses
+const combineStatuses = ({
```

Comment:
Do we need to care about old statuses? Won't we always want to use the latest value?

Reply:
Consider this case:
1. DescribeAsset succeeds
`{ isLoading: false, isRefetching: false, isSuccess, true, isError, false, errors: [] }`
2. GetLatest state fails
`{ isLoading: false, isRefetching: false, isSuccess, false, isError, true, errors: [<Error1>] }`
3. GetLatest type succeeds
`{ isLoading: false, isRefetching: false, isSuccess, false, isError, true, errors: [<Error1>] }`
4. GetLatest source fails
`{ isLoading: false, isRefetching: false, isSuccess, false, isError, true, errors: [<Error1>, <Error2>] }`

The new status at every step needs to resolve based on the old status to compile all of the errors and have the most accurate status.

We will still return AlarmData but it will have errors and isSuccess is false even if some queries were successful. Without caring about the old status we wouldn't have an accurate status for all of the queries.